tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -32,18 +40,41 @@ class BaseSuite extends munit.FunSuite with Assertions {
 
   override def munitTimeout: Duration = Duration("10min")
 
-  // NOTE(olafur): always ignore flak test failures.
+  // NOTE(olafur): always ignore flaky test failures.
   override def munitFlakyOK = true
-  override def munitRunTest(
-      options: TestOptions,
-      body: => Any
-  ): Any = {
-    if (Properties.isWin && options.tags(FlakyWindows)) {
-      munitFlaky(options, body)
-    } else {
-      super.munitRunTest(options, body)
-    }
-  }
+
+  case class Rerun(count: Int) extends munit.Tag("Rerun")
+
+  override def munitTestTransforms: List[TestTransform] =
+    super.munitTestTransforms ++ List(
+      new TestTransform("Rerun", { test =>
+        val rerunCount = test.tags
+          .collectFirst { case Rerun(n) => n }
+          .getOrElse(1)
+        if (rerunCount == 1) test
+        else {
+          test.withBody(() => {
+            Future.sequence(1.to(rerunCount).map(_ => test.body()).toList)
+          })
+        }
+      }),
```

Comment:
is this a new tag we want to use? Do we use it anywhere?

Reply:
We don't actually use it anywhere, but the old implementation of `FlakyTest` had the ability to pass in a number for max retries. Since I removed that I wanted to replace it with something that offered the same functionality. Do you think I should just remove it?

Reply:
I personally think so, since it's unused. We can easily add it back if we need it